Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1069,31 +1215,156 @@ public class ProgramBuilder {
-    public func createObject(with initialProperties: [String: Variable]) -> Variable {
-        // CreateObject expects sorted property names
-        var propertyNames = [String](), propertyValues = [Variable]()
-        for (k, v) in initialProperties.sorted(by: { $0.key < $1.key }) {
-            propertyNames.append(k)
-            propertyValues.append(v)
+    public struct Property {
+        var name: String
+        var value: Variable
+
+        init(_ name: String, _ value: Variable){
+            self.name = name
+            self.value = value
         }
-        return perform(CreateObject(propertyNames: propertyNames), withInputs: propertyValues).output
     }
 
-    @discardableResult
-    public func createArray(with initialValues: [Variable]) -> Variable {
-        return perform(CreateArray(numInitialValues: initialValues.count), withInputs: initialValues).output
+    public struct ComputedProperty {
+        var computedVal: Variable
+        var value: Variable
+
+        init(_ computedVal: Variable, _ value: Variable){
+            self.computedVal = computedVal
+            self.value = value
+        }
+    }
+
+    public struct SpreadProperty {
+        var value: Variable
+
+        init(_ value: Variable){
+            self.value = value
+        }
+    }
+
+    public struct GetterProperty {
+        public typealias GetterFunctionGenerator = () -> ()
+        var name: String
+        var body: GetterFunctionGenerator
+
+        init(_ name: String, _ body: @escaping GetterFunctionGenerator) {
+            self.name = name
+            self.body = body
+        }
+    }
+
+    public struct ComputedGetterProperty {
+        public typealias GetterFunctionGenerator = () -> ()
+        var computedVal: Variable
+        var body: GetterFunctionGenerator
+
+        init(_ computedVal: Variable, _ body: @escaping GetterFunctionGenerator) {
+            self.computedVal = computedVal
+            self.body = body
+        }
+    }
+
+    public struct SetterProperty {
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        var name: String
+        var body: SetterFunctionGenerator
+
+        init(_ name: String, _ body: @escaping SetterFunctionGenerator) {
+            self.name = name
+            self.body = body
+        }
+    }
+
+    public struct ComputedSetterProperty {
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        var computedVal: Variable
+        var body: SetterFunctionGenerator
+
+        init(_ computedVal: Variable, _ body: @escaping SetterFunctionGenerator) {
+            self.computedVal = computedVal
+            self.body = body
+        }
+    }
+
+    public struct ObjectBuilder {
+        public typealias GetterFunctionGenerator = () -> ()
+        public typealias SetterFunctionGenerator = (Variable) -> ()
+        fileprivate var properties: [Any] = []
+
+        public mutating func addProperty(_ name: String, v: Variable) {
```

Comment:
Nit: let's maybe use this signature here and below `(_ name: String, withValue v: Variable)`